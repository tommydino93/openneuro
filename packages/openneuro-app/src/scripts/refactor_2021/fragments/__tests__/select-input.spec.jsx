import React from 'react'
import { shallow } from 'enzyme'
import SelectInput from '../select-input'

describe('SelectInput', () => {
  it('handles display as disabled', () => {
    const wrapper = shallow(
      <SelectInput
        {...{
          name: 'Expected name',
          label: 'Expected label',
          value: 'hi',
          options: [{ value: 'hi' }, { value: 'bye' }],
          disabled: true,
          annotated: true,
        }}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('handles display as not disabled', () => {
    const wrapper = shallow(
      <SelectInput
        {...{
          name: 'Expected name',
          label: 'Expected label',
          value: 'hi',
          options: [{ value: 'hi' }, { value: 'bye' }],
          disabled: false,
          annotated: false,
        }}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('handles boolean display', () => {
    const wrapper = shallow(
      <SelectInput
        {...{
          name: 'Expected name',
          label: 'Expected label',
          value: true,
          options: [
            { value: true, text: 'true-label' },
            { value: false, text: 'false-label' },
          ],
          hasBooleanValues: true,
          disabled: true,
          annotated: true,
        }}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('handles boolean values', () => {
    let _, value
    const onChange = (...args) => ([_, value] = args)
    const wrapper = shallow(
      <SelectInput
        {...{
          name: 'Expected name',
          label: 'Expected label',
          value: true,
          options: [
            { value: true, text: 'true-label' },
            { value: false, text: 'false-label' },
          ],
          hasBooleanValues: true,
          disabled: true,
          annotated: true,
          onChange,
        }}
      />,
    )
    const event = {
      preventDefault() {
        null
      },
      target: { value: 'false' },
    }
    wrapper.find('styledSelect[name="Expected name"]').simulate('change', event)
    expect(value).toBe(false)
  })
})
