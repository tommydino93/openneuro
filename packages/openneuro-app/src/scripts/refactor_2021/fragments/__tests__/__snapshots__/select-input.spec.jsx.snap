// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectInput handles boolean display 1`] = `
<Fragment>
  <Styled(div)>
    <Styled(label)
      hasValue={true}
      htmlFor="Expected name"
    >
      Expected label
    </Styled(label)>
    <Styled(i)
      className="fa fa-asterisk"
    />
    <Styled(div)>
      true
      <SelectIcon />
    </Styled(div)>
    <styledSelect
      disabled={true}
      name="Expected name"
      onChange={[Function]}
      value="true"
    >
      <Styled(option)
        disabled={true}
        hidden={true}
        value=""
      />
      <Styled(option)
        key="0"
        value={true}
      >
        true-label
      </Styled(option)>
      <Styled(option)
        key="1"
        value={false}
      >
        false-label
      </Styled(option)>
    </styledSelect>
  </Styled(div)>
  <Styled(div)>
    <TextInput
      disabled={true}
      label="Other Expected label"
      name="Expected name"
      value="true"
    />
  </Styled(div)>
</Fragment>
`;

exports[`SelectInput handles display as disabled 1`] = `
<Fragment>
  <Styled(div)>
    <Styled(label)
      hasValue={true}
      htmlFor="Expected name"
    >
      Expected label
    </Styled(label)>
    <Styled(i)
      className="fa fa-asterisk"
    />
    <Styled(div)>
      hi
      <SelectIcon />
    </Styled(div)>
    <styledSelect
      disabled={true}
      name="Expected name"
      onChange={[Function]}
      value="hi"
    >
      <Styled(option)
        disabled={true}
        hidden={true}
        value=""
      />
      <Styled(option)
        key="0"
        value="hi"
      >
        hi
      </Styled(option)>
      <Styled(option)
        key="1"
        value="bye"
      >
        bye
      </Styled(option)>
    </styledSelect>
  </Styled(div)>
  <Styled(div)>
    <TextInput
      disabled={true}
      label="Other Expected label"
      name="Expected name"
      value="hi"
    />
  </Styled(div)>
</Fragment>
`;

exports[`SelectInput handles display as not disabled 1`] = `
<Fragment>
  <Styled(div)>
    <Styled(label)
      hasValue={true}
      htmlFor="Expected name"
    >
      Expected label
    </Styled(label)>
    <Styled(div)>
      hi
      <SelectIcon />
    </Styled(div)>
    <styledSelect
      disabled={false}
      name="Expected name"
      onChange={[Function]}
      value="hi"
    >
      <Styled(option)
        disabled={true}
        hidden={true}
        value=""
      />
      <Styled(option)
        key="0"
        value="hi"
      >
        hi
      </Styled(option)>
      <Styled(option)
        key="1"
        value="bye"
      >
        bye
      </Styled(option)>
    </styledSelect>
  </Styled(div)>
  <Styled(div)>
    <TextInput
      disabled={false}
      label="Other Expected label"
      name="Expected name"
      value="hi"
    />
  </Styled(div)>
</Fragment>
`;
